src/libhirte/common/cfg.c

```diff
@@ -106,23 +100,31 @@ int cfg_load_complete_configuration(
 
         if (default_config_file != NULL) {
                 result = cfg_load_from_file(config, default_config_file);
-                if (result != 0) {
-                        return result;
+                if (result < 0) {
+                        fprintf(stderr,
+                                "Error loading default configuration file '%s': '%s'.\n",
+                                default_config_file,
+                                strerror(-result));
                 }
         }
 
         if (custom_config_file != NULL) {
                 result = cfg_load_from_file(config, custom_config_file);
-                // if custom config file doesn't exist, continue
-                if (result != 0 && result != -ENOENT) {
-                        return result;
```

Comment:
Agreed. I'll add the return again. The print I'd keep here, though, since we can't determine on the caller-side for which file the loading failed - and having the file name I consider quite helpful.